### PR TITLE
Allow direct configuration without config file using mybatis-spring 1.3.0

### PR DESCRIPTION
**Pull request by kazuki43zoo**
I've fixed to support https://github.com/mybatis/spring/pull/108 on Spring Boot.
What do you think ?
Please review this.

### CODE REVIEW

**Comment by eddumelendez** [comment]
Looks ok, but we will have to wait until the release of the next version before to merge it. thanks @kazuki43zoo

File: mybatis-spring-boot-autoconfigure/src/main/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfiguration.java [hunk 1]
```diff
@@ -121,10 +126,19 @@ public SqlSessionFactory sqlSessionFactory(DataSource dataSource) throws Excepti
 			factory.setTypeHandlersPackage(this.properties.getTypeHandlersPackage());
 			factory.setMapperLocations(this.properties.resolveMapperLocations());
 		}
-
 		return factory.getObject();
 	}
 
+	/**
+	 * @since 1.1.0
```
Comment (eddumelendez) [inline mybatis-spring-boot-autoconfigure/src/main/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfiguration.java]: avoid javadocs in this place

File: mybatis-spring-boot-autoconfigure/src/main/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfiguration.java [hunk 2]
```diff
@@ -121,10 +126,19 @@ public SqlSessionFactory sqlSessionFactory(DataSource dataSource) throws Excepti
...
+	 */
+	@Bean
+	@ConditionalOnMissingBean
+	@ConfigurationProperties(prefix = MybatisProperties.MYBATIS_PREFIX + ".configuration")
+	public org.apache.ibatis.session.Configuration myBatisConfiguration() {
+		return new org.apache.ibatis.session.Configuration();
```
Comment (eddumelendez) [inline mybatis-spring-boot-autoconfigure/src/main/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfiguration.java]: Why not move it to MybatisProperties?

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java [hunk 3]
```diff
@@ -164,6 +166,27 @@ public void testWithDatabaseIdProvider() {
 		assertEquals("h2", this.context.getBean(SqlSessionFactory.class).getConfiguration().getDatabaseId());
 	}
 
+	@Test
+	public void testMyBatisConfigurationUsingConfigurationProperties() {
```
Comment (eddumelendez) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: add test with the default configuration

File: mybatis-spring-boot-autoconfigure/src/main/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfiguration.java (hunk 2 above)
Comment (kazuki43zoo) [inline mybatis-spring-boot-autoconfigure/src/main/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfiguration.java]: I doesn't have reasonable reason.
Therefor, i moved to `MybatisProperties` via 5670187f7b483125be4f06e145cf806ef36143e0.

File: mybatis-spring-boot-autoconfigure/src/main/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfiguration.java (hunk 1 above)
Comment (kazuki43zoo) [inline mybatis-spring-boot-autoconfigure/src/main/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfiguration.java]: fix via 5670187f7b483125be4f06e145cf806ef36143e0.

File: mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java (hunk 3 above)
Comment (kazuki43zoo) [inline mybatis-spring-boot-autoconfigure/src/test/java/org/mybatis/spring/boot/autoconfigure/MybatisAutoConfigurationTest.java]: Add minimum assertion on `testDefaultConfiguration()` via 5670187f7b483125be4f06e145cf806ef36143e0.
Should i added a new test case ?

**Comment by kazuki43zoo** [comment]
@eddumelendez , thanks for review comments.

I've fix your review comment and write a my opinions.
Please review again.

**Comment by kazuki43zoo** [comment]
Move to #47 .

**Pull request closed without merging**